### ISSUE TITLE
wazuh-modulesd

### ISSUE THREAD

**Issue by buzzzo** [wazuh/wazuh#4883]
Centos 7
Wazuh 12.0 12.1 

As soon as wazuh-manager start wazuh-modulesd died with:
wazuh-modulesd[2557]: segfault at 0 ip 00007f66ab159c70 sp 00007f66aa602978 error 4 in libc-2.17.so[7f66ab019000+1c3000]

**Comment by buzzzo** [comment]
running wazuh-modulesd -d -f  could be noted that segfault everytime at this point:

2020/04/08 23:58:21 wazuh-modulesd:download[2930] wm_download.c:213 at wm_download_dispatch(): DEBUG: Downloading 'https://access.redhat.com/labs/securitydataapi/cve.json?after=2010-01-01&per_page=1000&page=4' to '/var/ossec/tmp/vuln-temp-fitted'

**Comment by victoryure** [comment]
> Centos 7
> Wazuh 12.0 12.1
> 
> As soon as wazuh-manager start wazuh-modulesd died with:
> wazuh-modulesd[2557]: segfault at 0 ip 00007f66ab159c70 sp 00007f66aa602978 error 4 in libc-2.17.so[7f66ab019000+1c3000]

Same thing here.

**Comment by punapk** [comment]
Hello, same here.

I suspect that redhat's api feed has changed format or something like that
running  wazuh-moduled -df

`wm_vuln_detector.c:2721 at wm_vuldet_json_rh_parser(): DEBUG: (5493): The version of 'openshift-service-mesh/kiali-rhel7-operator:1.0.12-2' could not be extracted.`

I noticed that when i reduced the year to synchronize from (default:2010) to something more recent like 2020. It doesn't segfault but i still have some error about `wm_vuln_detector.c:2721 at wm_vuldet_json_rh_parser(): DEBUG: (5493): The version of 'openshift3/mediawiki:v3.11.188-2' could not be extracted.`
Maybe im wrong...

Hope it helps

**Comment by MiguelCasaresRobles** [comment]
Hello @punapk , @victoryure , @buzzzo ,

the problem may be related to a bug in vulnerability detector, the Redhat feed database made a change and that caused the error in our module.

To detect it you may check if the wazuh-modulesd process is running:

`ps aux | grep ossec`

Otherwise, you need to edit your /var/ossec/etc/ossec.conf file on the Wazuh manager side. For Wazuh versions prior to 3.11 inside your vulnerability configuration do this for the RedHat feed:
```
<feed name="redhat">
<disabled>yes</disabled>
</feed>
```

For Wazuh versions after 3.11 inside your vulnerability configuration do this for the RedHat feed:
```
<provider name="redhat">
<enabled>no</enabled>
</provider>

```
Then restart your Wazuh manager:

`systemctl restart wazuh-manager`

Please make sure the manager is properly working afterward.

We released the fix in a new version yesterday, you may check it out here: https://documentation.wazuh.com/3.12/release-notes/index.html

This is the issue: https://github.com/wazuh/wazuh/issues/4884

Apologies for the inconvenience and do not hesitate to contact us if you have further problems.

Regards,

Miguel Casares

**Comment by vikman90** [comment]
Hi @buzzzo @victoryure @punapk,

This bug should have been fixed by #4885, just as @MiguelCasaresRobles mentioned.

Let me close this issue. Should this problem persists, don't hesitate to reopen it.

Best regards.
